gpuflow/dataplane/gpu/cuda/cuda_lpm_factory.cu

```diff
@@ -110,11 +111,10 @@ int IPv6LPMFactory::AddLPMRule(uint8_t *ipv6_address, uint8_t depth, uint8_t nex
   unsigned long int end;
   if (depth == 24) {
     // The depth is exactly 24
-
     start = ipv6_address[0] << 16 | ipv6_address[1] << 8 | ipv6_address[2];
     end = ipv6_address[0] << 16 | ipv6_address[1] << 8 | ipv6_address[2] + 1;
```

Comment:
It seems like only when ipv6_address[0], ipv6_address[1] and ipv6_address[2] all equal 255 that would cause the problem of overflow?

Reply:
In this case, when `ipv6_address[2] == 255`, it will cause unexpected result with no braces?
In other cases, just make sure the extreme cases are checked is fine.